tools/postgres/diff_schema.js

```diff
@@ -0,0 +1,160 @@
+const scriptdir = __dirname;
+const scriptName = __filename;
+let p1;
+let p2;
+let dir1;
+let dir2;
+
+function dump(parameter) {
+  if (parameter === "DB") {
+    POSTGRES_URL = ORIGINAL_POSTGRES_URL; // this environment variable is passed to dump_schema.sh
+    return dumpToFolder();
+  } else {
+    const { dbName, dbHost } = initTmpDB();
+    console.log(`Creating DB ${dbName}`);
+    execSync(`psql -U postgres -h ${dbHost} -c "CREATE DATABASE ${dbName};"`, {
+      env: { PGPASSWORD: "postgres" },
+    });
+    try {
+      loadDataIntoDB(parameter, dbHost, dbName);
+      return dumpToFolder();
+    } catch (err) {
+      console.log(err);
+      process.exit(1);
+    } finally {
+      console.log(`CLEANUP: DROP DATABASE ${dbName}`);
+      execSync(`psql -U postgres -h ${dbHost} -c "DROP DATABASE ${dbName};"`, {
+        env: { PGPASSWORD: "postgres" },
+      });
+    }
+  }
+  return null;
+}
+
+function initTmpDB() {
+  const tempDbName = generateRandomName();
+  const postgresDirname = path.dirname(ORIGINAL_POSTGRES_URL);
+  POSTGRES_URL = `${postgresDirname}/${tempDbName}`; // this environment variable is passed to dump_schema.sh
+  const dbName = execSync(`${scriptdir}/db_name.sh`, { env: { POSTGRES_URL } })
+    .toString()
+    .trim(); // "trim" to remove the line break
+  if (dbName !== tempDbName) {
+    console.log("Wrong dbName");
+    process.exit(1);
+  }
+  const dbHost = execSync(`${scriptdir}/db_host.sh`, { env: { POSTGRES_URL } })
+    .toString()
+    .trim();
+  return {
+    dbName,
+    dbHost,
+  };
+}
+
+function loadDataIntoDB(parameter, dbHost, dbName) {
+  const fileNames = glob.sync(parameter);
+  const concatenateFileNames = fileNames.map(name => `-f ${name}`).join(" ");
+  execSync(
+    `psql -U postgres -h ${dbHost} --dbname="${dbName}" -v ON_ERROR_STOP=ON -q ${concatenateFileNames}`,
+    { env: { PGPASSWORD: "postgres" } },
+  );
+}
+
+function dumpToFolder() {
+  const tmpDir = tmp.dirSync();
+  try {
+    execSync(`${scriptdir}/dump_schema.sh ${tmpDir.name}`, { env: { POSTGRES_URL } });
+  } catch (err) {
+    console.log(`CLEANUP: remove ${tmpDir.name}`);
+    rimraf.sync(tmpDir.name);
+    process.exit(1);
+  }
+  return tmpDir.name;
+}
+
+function generateRandomName() {
+  const random = randomstring.generate({
+    length: 8,
+    charset: "alphanumeric",
+    capitalization: "lowercase",
+  });
+  return `wk_tmp_${random}`;
+}
+
+function sortFile(fileName) {
+  try {
+    const content = fs.readFileSync(fileName);
+    let lines = content.toString().split("\n");
+    lines = lines.sort();
+    fs.writeFileSync(fileName, lines.join("\n"));
+  } catch (err) {
+    console.log(`FAILED to sort file ${fileName}`);
+    process.exit(1);
+  }
+}
+
+program
+  .version("0.1.0", "-v, --version")
+  .arguments("<parameter1> <parameter2>")
+  .action((parameter1, parameter2) => {
+    p1 = parameter1;
+    p2 = parameter2;
+  });
+
+if (process.argv.length !== 4) {
```

Comment:
Doesn't `commander` do this check? It can also print a usage message.

Reply:
Commander only shows the usage-message when you call it with `--help` but not if you forget a parameter